src/recipe/recipe.service.spec.ts

```diff
@@ -1,18 +1,188 @@
+          },
+        ]);
+      }),
+
+      update: jest
+        .fn()
+        .mockImplementation(
+          (request: { where: { id: number }; data: MockRecipeRequest }) => {
+            return Promise.resolve({
+              id: request.where.id,
+              createdAt: Date.now(),
+              ...request.data,
+            });
+          },
+        ),
+
+      delete: jest.fn().mockImplementation((_request) => {
+        return Promise.resolve();
+      }),
+    },
+  };
 
   beforeEach(async () => {
     const module: TestingModule = await Test.createTestingModule({
-      providers: [RecipeService],
+      providers: [
+        RecipeService,
+        {
+          provide: PrismaService,
+          useValue: mockPrismaService,
+        },
+      ],
     }).compile();
 
-    service = module.get<RecipeService>(RecipeService);
+    recipeService = module.get<RecipeService>(RecipeService);
   });
 
   it('should be defined', () => {
-    expect(service).toBeDefined();
+    expect(recipeService).toBeDefined();
+  });
+
+  describe('CreateRecipe', () => {
+    it('should create recipe', async () => {
+      //given
+      const request = {
+        title: faker.word.noun(),
+        description: faker.lorem.text(),
+        ingredients: faker.lorem.word(5),
+        preparation: faker.lorem.lines(3),
+      };
+
+      //when
+      const createdRecipe = await recipeService.createRecipe(request);
+
+      //then
+      expect(createdRecipe).toEqual(mockRecipe);
+    });
+  });
+
+  describe('FetchRecipe', () => {
+    it('should fetch recipe', async () => {
+      //given
+      const id = 777;
+
+      //when
+      const fetchedRecipe = await recipeService.fetchRecipe(id);
+
+      //then
+      expect(fetchedRecipe).toEqual({
+        id: expect.any(Number),
+        createdAt: expect.any(Number),
+        title: expect.any(String),
+        description: expect.any(String),
+        ingredients: expect.any(String),
+        preparation: expect.any(String),
+      });
+    });
+  });
+
+  describe('FetchAllRecipes', () => {
+    it('should fetch all recipes', async () => {
+      //given
+
+      //when
+      const fetchedAllRecipes = await recipeService.fetchAllRecipes();
+
+      //then
+      expect(fetchedAllRecipes).toEqual(expect.arrayContaining([mockRecipe]));
+    });
+  });
+
+  describe('UpdateRecipe', () => {
+    it('should update recipe by given id', async () => {
+      //given
+      const id = 11;
+      const payload = {
+        title: faker.word.noun(),
+        description: faker.lorem.text(),
+        ingredients: faker.lorem.word(5),
+        preparation: faker.lorem.lines(3),
+      };
+
+      //when
+      const updatedRecipe = await recipeService.updateRecipe(id, payload);
+
+      //then
+      expect(updatedRecipe).toEqual(mockRecipe);
+    });
+  });
+
+  describe('DeleteRecipe', () => {
+    it('should delete recipe by given id', async () => {
+      //given
+      const id = 37;
```

Comment:
if id does not play important role here, I'd set a random id each time (you can use faker or Math.random for example)